### PR TITLE
TASK-319 Export number of daytime employees information

### PR DESCRIPTION
**Pull request by VVlovsky**
Zwrócono funkcjonalność exportu z pracownikami dziennymi, parser zmodyfikowano dla obsługi odpowiedniej sytuacji, zaimplementowana obsługa błędów wczytywania.
image

### CODE REVIEW

**Comment by pectom** [comment]
@kaplonpaulina czekamy na review ;)

File: src/logic/schedule-parser/extra-workers-info.parser.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  }
+
+  private logLoadFileError(msg: string): void {
+    this._parseErrors.push({
+      kind: InputFileErrorCode.LOAD_FILE_ERROR,
+      message: msg,
+    });
+  }
+
+  private generateChildren(raw?: string[][]): number[] {
```
Comment (kaplonpaulina) [inline src/logic/schedule-parser/extra-workers-info.parser.ts:28]: rename

File: src/logic/schedule-parser/children-info.parser.ts [hunk 2]
```diff
@@ -30,20 +30,26 @@ export class ChildrenInfoParser implements ChildrenInfoProvider {
...
-        "Brak informacji o liczbie dzieci. Przyjęto, że w każdym dniu liczba dzieci wynosi " +
-          this.DEFAULT_CHILDREN_NUMBER
-      );
+    let childrenRow;
 
-      const N = this.metaData.dayCount;
-      const children = Array(N);
-      let i = 0;
+    if (!raw || raw.length !== 1) {
+      if (raw && raw.length === 3 && raw[2][0].toLowerCase() === "dzieci") {
```
Comment (kaplonpaulina) [inline src/logic/schedule-parser/children-info.parser.ts:36]: redundant condition
```suggestion
      if (raw.length === 3 && raw[2][0].toLowerCase() === "dzieci") {
```
check if raw[2] exist before getting first element

File: src/logic/schedule-parser/schedule.parser.ts [hunk 3]
```diff
@@ -43,6 +44,12 @@ export class ScheduleParser implements ScheduleProvider {
       if (r && r.length !== 0 && r[0].length !== 0) {
```
Comment (kaplonpaulina) [inline src/logic/schedule-parser/schedule.parser.ts:44]: it shouldn' work now?

File: src/logic/schedule-parser/metadata.parser.ts [hunk 4]
```diff
@@ -46,7 +46,7 @@ export class MetaDataParser extends MetadataProvider {
   }
 
   private extractMetadata(raw: string[][]): number {
-    if (raw.length !== 1) {
+    if (raw.length !== 1 && raw.length !== 3) {
```
Comment (kaplonpaulina) [inline src/logic/schedule-parser/metadata.parser.ts:49]: ?? do we want 1 row or 3 rows ?? - a bad condition in general

File: src/logic/schedule-parser/schedule.parser.ts [hunk 5]
```diff
@@ -35,6 +35,7 @@ export class ScheduleParser implements ScheduleProvider {
     let children;
     let nurses;
     let babysiter;
+    let dayWorkers;
 
     const metadataRaw = rawSchedule.find((r) => r[0][0].toLowerCase().trim() === "dni miesiąca");
```
Comment (kaplonpaulina) [inline src/logic/schedule-parser/schedule.parser.ts:40]: you change the rawSchedule, we have to change also this,
metadataRaw was before [["dni miesiąca" , 1 ,2 3,4 ...]]
now:
[["dni miesiąca ...],
["pracownicy dzienni ...],
["dzieci" ...]]

this should be adressed

File: src/logic/schedule-parser/schedule.parser.ts [hunk 6]
```diff
@@ -43,6 +44,12 @@ export class ScheduleParser implements ScheduleProvider {
       if (r && r.length !== 0 && r[0].length !== 0) {
         const rKey = r[0][0].toLowerCase().trim();
         if (rKey === "dni miesiąca") {
+          if (r.length === 3) {
```
Comment (kaplonpaulina) [inline src/logic/schedule-parser/schedule.parser.ts:47]: what if only children are present?

File: src/logic/schedule-parser/schedule.parser.ts [hunk 7]
```diff
@@ -35,6 +35,7 @@ export class ScheduleParser implements ScheduleProvider {
     let children;
     let nurses;
     let babysiter;
+    let dayWorkers;
```
Comment (kaplonpaulina) [inline src/logic/schedule-parser/schedule.parser.ts:38]: extraWorkers - let's keep the naming convention

File: src/logic/schedule-parser/schedule.parser.ts (hunk 6 above)
Comment (kaplonpaulina) [inline src/logic/schedule-parser/schedule.parser.ts:47]: I don't really like this solution, 
I am making MR to your branch with suggestion how I would solve this if we want to have all "dni miesiąca", "pracownicy dzienni", "dzieci" in one array

Comment (kaplonpaulina) [reply src/logic/schedule-parser/schedule.parser.ts:47]: Please let me know if you agree or think it will make things more complicated

**Review by kaplonpaulina — CHANGES_REQUESTED** [review]
Please review the suggestion MR and let me know what you think about it, 
I believe changes in the rawShedule should be addressed, it doesn't have to be done as I suggested

File: src/logic/schedule-parser/schedule.parser.ts (hunk 6 above)
Comment (kaplonpaulina) [reply src/logic/schedule-parser/schedule.parser.ts:47]: https://github.com/Project-Summer-AI-Lab-Glider/nurse-scheduling-problem-frontend/pull/268

**Review by kaplonpaulina — APPROVED** [review]

**Pull request merged**